Add the ability to navigate with swipe gesture

pan93412 opened the pull request:
On macOS, add an API to enable or disable backward and forward navigation gestures.

### What kind of change does this PR introduce?
<!-- Check at least one. If you are introducing a new binding, you must reference an issue where this binding has been proposed, discussed and approved by the maintainers. -->

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
<!-- If yes, please describe the impact and migration path for existing applications in an attached issue. -->

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Demo video
https://user-images.githubusercontent.com/28441561/202171447-b6cb4750-0326-466f-8e19-3371d996784b.mov

### Other information
The documentation of this function can be seen on <https://developer.apple.com/documentation/webkit/wkwebview/1414995-allowsbackforwardnavigationgestu>.

amrbashir commented on examples/with_forward_gesture.rs:
```diff
@@ -0,0 +1,36 @@
+// Copyright 2020-2022 Tauri Programme within The Commons Conservancy
```
Example is not needed, please remove it.

amrbashir commented on .changes/allow-back-forward-gesture.md:
```diff
@@ -0,0 +1,11 @@
+"wry": minor
+---
+
+Add the ability to navigate with swipe gesture
+
+This switch is for WebView on macOS, enabling us to swipe horizontally to trigger backward and forward page navigation.
+
+This commit also includes a worked example for it.
+
+For details, see <https://developer.apple.com/documentation/webkit/wkwebview/1414995-allowsbackforwardnavigationgestu>
```
```suggestion
On macOS, add an API to enable or disable backward and forward navigation gestures.
```

pan93412 committed:
Add the ability to navigate with swipe gesture

On macOS, add an API to enable or disable
backward and forward navigation gestures.

For details, see <https://developer.apple.com/documentation/webkit/wkwebview/1414995-allowsbackforwardnavigationgestu>

Signed-off-by: pan93412 <pan93412@gmail.com>

amrbashir commented on .changes/allow-back-forward-gesture.md:
```diff
@@ -0,0 +1,9 @@
+"wry": minor
+---
+
+Add the ability to navigate with swipe gesture
+
+On macOS, add an API to enable or disable backward and forward navigation gestures.
+
+For details, see <https://developer.apple.com/documentation/webkit/wkwebview/1414995-allowsbackforwardnavigationgestu>
```
```suggestion
On macOS, add an API to enable or disable backward and forward navigation gestures.
```

amrbashir committed:
Update .changes/allow-back-forward-gesture.md

amrbashir reviewed (approved):
Thanks

daveycodez commented:
How can I enable this on a new Tauri app for macOS

FabianLars commented:
I don't think this was ever exposed in Tauri but you can use this code https://github.com/tauri-apps/wry/pull/757/files#diff-71837c0d08219b9f7bbb5b131c1eba092f7fa5fe6da910b8fd4254b680a17278 in tauri's with_webview method.

daveycodez commented:
> I don't think this was ever exposed in Tauri but you can use this code https://github.com/tauri-apps/wry/pull/757/files#diff-71837c0d08219b9f7bbb5b131c1eba092f7fa5fe6da910b8fd4254b680a17278 in tauri's with_webview method.

I'm not totally sure how to use Tauri's with_webview method? I'm pretty new to Rust

FabianLars commented:
There's a small example in the docs https://docs.rs/tauri/latest/tauri/webview/struct.WebviewWindow.html#method.with_webview - i think it may be enough to change the `view.setPageZoom()` line to `view.allowsBackForwardNavigationGestures(false)`

daveycodez commented:
> There's a small example in the docs https://docs.rs/tauri/latest/tauri/webview/struct.WebviewWindow.html#method.with_webview - i think it may be enough to change the `view.setPageZoom()` line to `view.allowsBackForwardNavigationGestures(false)`

Thanks! This worked.

Here is my /src-tauri/src/lib.rs for anyone who runs into this

```rust
use tauri::Manager;

#[cfg_attr(mobile, tauri::mobile_entry_point)]
pub fn run() {
    tauri::Builder::default()
        .plugin(tauri_plugin_opener::init())
        .plugin(tauri_plugin_deep_link::init())
        .plugin(tauri_plugin_http::init())
        .setup(|app| {
            if cfg!(debug_assertions) {
                app.handle().plugin(
                    tauri_plugin_log::Builder::default()
                        .level(log::LevelFilter::Info)
                        .build(),
                )?;
            }

            let main_webview = app.get_webview_window("main").unwrap();
            let _ = main_webview.with_webview(|webview| {
                #[cfg(target_os = "macos")]
                unsafe {
                    let view: &objc2_web_kit::WKWebView = &*webview.inner().cast();
                    view.setAllowsBackForwardNavigationGestures(true);
                }
            });

            Ok(())
        })
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

daveycodez commented:
> There's a small example in the docs https://docs.rs/tauri/latest/tauri/webview/struct.WebviewWindow.html#method.with_webview - i think it may be enough to change the `view.setPageZoom()` line to `view.allowsBackForwardNavigationGestures(false)`

How can I make this work on iOS and Android as well? And Windows? It seems really complex. This really should be built into tauri.config
